diff --git a/CMakeLists.txt b/CMakeLists.txt
index fc61af7..b2b594e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -303,7 +303,7 @@ if (MAGMA_ENABLE_HIP)
     set(GPU_ARCH_FLAGS ${DEVCCFLAGS})
 
     #add_compile_options(${GPU_ARCH_FLAGS})
-    set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D__HIP_PLATFORM_AMD__" )
+    set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D__HIP_PLATFORM_AMD__ -DHIPBLAS_V2" )
     option(ROCM_CORE "Location of the rocm-core package")
     execute_process(COMMAND "${CMAKE_SOURCE_DIR}/tools/get-rocm-version.sh" "${ROCM_CORE}" OUTPUT_VARIABLE ROCM_VERSION OUTPUT_STRIP_TRAILING_WHITESPACE)
     message(STATUS "ROCM_VERSION=${ROCM_VERSION}")
diff --git a/Makefile b/Makefile
index 44880ab..afaab45 100644
--- a/Makefile
+++ b/Makefile
@@ -846,7 +846,7 @@ ifeq ($(BACKEND),cuda)
 d_ext := cu
 else ifeq ($(BACKEND),hip)
 d_ext := cpp
-CXXFLAGS += -D__HIP_PLATFORM_AMD__
+CXXFLAGS += -D__HIP_PLATFORM_AMD__ -DHIPBLAS_V2
 CXXFLAGS += -DROCM_VERSION=$(shell ./tools/get-rocm-version.sh)
 endif
 
diff --git a/include/magma_types.h b/include/magma_types.h
index d8dcf92..f36aa22 100644
--- a/include/magma_types.h
+++ b/include/magma_types.h
@@ -227,8 +227,6 @@ typedef double real_Double_t;
 
     /* double complex */
 
-    //typedef hipblasDoubleComplex magmaDoubleComplex;
-
     /* simple double complex definition that should be binary compatible with hipBLAS */
     typedef struct {
 
@@ -275,9 +273,6 @@ typedef double real_Double_t;
 
     /* float complex */
 
-    //typedef hipComplex magmaFloatComplex;
-    //typedef hipblasComplex magmaFloatComplex;
-
     /* basic definition of float complex that should be binary compatible with hipBLAS */
     typedef struct {
 
diff --git a/interface_cuda/blas_h_v2.cpp b/interface_cuda/blas_h_v2.cpp
index 5617031..474e1e3 100644
--- a/interface_cuda/blas_h_v2.cpp
+++ b/interface_cuda/blas_h_v2.cpp
@@ -118,10 +118,10 @@ magma_hgemm(
 		      hipblas_trans_const( transA ),
 		      hipblas_trans_const( transB ),
 		      int(m), int(n), int(k),
-		      (void*)&alpha, (void*)dA, HIPBLAS_R_16F, int(ldda),
-		      (void*)dB, HIPBLAS_R_16F, int(lddb),
-		      (void *)&beta,  (void*)dC, HIPBLAS_R_16F, int(lddc),
-		      HIPBLAS_R_16F,
+		      (void*)&alpha, (void*)dA, HIP_R_16F, int(ldda),
+		      (void*)dB, HIP_R_16F, int(lddb),
+		      (void *)&beta,  (void*)dC, HIP_R_16F, int(lddc),
+		      HIPBLAS_COMPUTE_16F,
 		      HIPBLAS_GEMM_DEFAULT);
     }
     else {
@@ -151,10 +151,10 @@ magma_hgemmx(
 		      hipblas_trans_const( transA ),
 		      hipblas_trans_const( transB ),
 		      int(m), int(n), int(k),
-		      (void*)&alpha, (void*)dA, HIPBLAS_R_16F, int(ldda),
-                                     (void*)dB, HIPBLAS_R_16F, int(lddb),
-		      (void*)&beta,  (void*)dC, HIPBLAS_R_32F, int(lddc),
-		      HIPBLAS_R_32F,
+		      (void*)&alpha, (void*)dA, HIP_R_16F, int(ldda),
+                                     (void*)dB, HIP_R_16F, int(lddb),
+		      (void*)&beta,  (void*)dC, HIP_R_32F, int(lddc),
+		      HIPBLAS_COMPUTE_32F,
 		      HIPBLAS_GEMM_DEFAULT);
     }
     else {
diff --git a/magmablas/zgemm_batched.cpp b/magmablas/zgemm_batched.cpp
index b0aec1c..73140b8 100644
--- a/magmablas/zgemm_batched.cpp
+++ b/magmablas/zgemm_batched.cpp
@@ -28,13 +28,13 @@
  *       */
 #ifdef PRECISION_z
   #ifdef MAGMA_HAVE_HIP
-    typedef hipblasDoubleComplex BackendFloat_t;
+    typedef hipDoubleComplex BackendFloat_t;
   #else
     typedef cuDoubleComplex BackendFloat_t;
   #endif
 #elif defined(PRECISION_c)
   #ifdef MAGMA_HAVE_HIP
-    typedef hipblasComplex BackendFloat_t;
+    typedef hipComplex BackendFloat_t;
   #else
     typedef cuFloatComplex BackendFloat_t;
   #endif
diff --git a/magmablas/zgetf2_kernels.cu b/magmablas/zgetf2_kernels.cu
index 5ebfeef..b162025 100644
--- a/magmablas/zgetf2_kernels.cu
+++ b/magmablas/zgetf2_kernels.cu
@@ -211,7 +211,7 @@ magma_izamax_native(
         hipblasGetPointerMode(queue->hipblas_handle(), &ptr_mode);
         hipblasSetPointerMode(queue->hipblas_handle(), CUBLAS_POINTER_MODE_DEVICE);
 
-        hipblasIzamax(queue->hipblas_handle(), length, (const hipblasDoubleComplex*)x, 1, (int*)(ipiv));
+        hipblasIzamax(queue->hipblas_handle(), length, (const hipDoubleComplex*)x, 1, (int*)(ipiv));
         magma_zpivcast<<< 1, 1, 0, queue->cuda_stream() >>>( ipiv );
 
         hipblasSetPointerMode(queue->hipblas_handle(), ptr_mode);
diff --git a/make.inc-examples/make.inc.hip-gcc-mkl b/make.inc-examples/make.inc.hip-gcc-mkl
index e566419..afdae21 100644
--- a/make.inc-examples/make.inc.hip-gcc-mkl
+++ b/make.inc-examples/make.inc.hip-gcc-mkl
@@ -112,12 +112,6 @@ ifeq ($(BACKEND),cuda)
     DEVCCFLAGS += -Xcompiler "$(FPIC)" -Xcompiler "$(FOPENMP)" -std=c++11
 else ifeq ($(BACKEND),hip)
     DEVCCFLAGS += $(FPIC) $(FOPENMP) -std=c++11
-    # check for older versions of ROCM
-    ifeq ($(shell hipconfig --version | cut -b -3),3.0)
-        $(info Building with HIP 3.0)
-        # they don't have hipblasComplex yet, so replace it manually
-        DEVCCFLAGS += -DhipblasComplex=hipComplex -DhipblasDoubleComplex=hipDoubleComplex
-    endif
 endif
 
 
diff --git a/make.inc-examples/make.inc.hip-gcc-openblas b/make.inc-examples/make.inc.hip-gcc-openblas
index df71f09..b257efa 100644
--- a/make.inc-examples/make.inc.hip-gcc-openblas
+++ b/make.inc-examples/make.inc.hip-gcc-openblas
@@ -116,12 +116,6 @@ ifeq ($(BACKEND),cuda)
     DEVCCFLAGS += -Xcompiler "$(FPIC)" -Xcompiler "$(FOPENMP)" -std=c++11
 else ifeq ($(BACKEND),hip)
     DEVCCFLAGS += $(FPIC) $(FOPENMP) -std=c++11
-    # check for older versions of ROCM
-    ifeq ($(shell hipconfig --version | cut -b -3),3.0)
-        $(info Building with HIP 3.0)
-        # they don't have hipblasComplex yet, so replace it manually
-        DEVCCFLAGS += -DhipblasComplex=hipComplex -DhipblasDoubleComplex=hipDoubleComplex
-    endif
 endif
 
 
diff --git a/sparse/blas/magma_z_blaswrapper.cpp b/sparse/blas/magma_z_blaswrapper.cpp
index 598968d..7b9b74c 100644
--- a/sparse/blas/magma_z_blaswrapper.cpp
+++ b/sparse/blas/magma_z_blaswrapper.cpp
@@ -13,15 +13,6 @@
 
 #define PRECISION_z
 
-/* For hipSPARSE, they use a separate complex type than for hipBLAS */
-#if defined(MAGMA_HAVE_HIP)
-  #ifdef PRECISION_z
-    #define hipblasDoubleComplex hipDoubleComplex
-  #elif defined(PRECISION_c)
-    #define hipblasComplex hipComplex
-  #endif
-#endif
-
 #if CUDA_VERSION >= 12000
   #define CUSPARSE_CSRMV_ALG2 CUSPARSE_SPMV_CSR_ALG2
   #define CUSPARSE_CSRMV_ALG1 CUSPARSE_SPMV_CSR_ALG1
diff --git a/sparse/blas/magma_zmatrixtools_gpu.cu b/sparse/blas/magma_zmatrixtools_gpu.cu
index 4f6b8ad..3379d89 100644
--- a/sparse/blas/magma_zmatrixtools_gpu.cu
+++ b/sparse/blas/magma_zmatrixtools_gpu.cu
@@ -14,11 +14,6 @@
 
 #define SWAP(a, b)  { tmp = a; a = b; b = tmp; }
 
-/* For hipSPARSE, they use a separate complex type than for hipBLAS */
-#ifdef MAGMA_HAVE_HIP
-  #define hipblasDoubleComplex hipDoubleComplex
-#endif
-
 
 
 __global__ void 
diff --git a/sparse/blas/magma_ztrisolve.cpp b/sparse/blas/magma_ztrisolve.cpp
index acc1345..ef4c2bb 100644
--- a/sparse/blas/magma_ztrisolve.cpp
+++ b/sparse/blas/magma_ztrisolve.cpp
@@ -13,15 +13,6 @@
 
 #define PRECISION_z
 
-/* For hipSPARSE, they use a separate complex type than for hipBLAS */
-#if defined(MAGMA_HAVE_HIP)
-  #ifdef PRECISION_z
-    #define hipblasDoubleComplex hipDoubleComplex
-  #elif defined(PRECISION_c)
-    #define hipblasComplex hipComplex
-  #endif
-#endif
-
 magma_int_t magma_ztrisolve_analysis(magma_z_matrix M, magma_solve_info_t *solve_info, bool upper_triangular, bool unit_diagonal, bool transpose, magma_queue_t queue)
 {
     magma_int_t info = 0;
diff --git a/sparse/blas/zilu.cpp b/sparse/blas/zilu.cpp
index 44ec1fc..9d50de1 100644
--- a/sparse/blas/zilu.cpp
+++ b/sparse/blas/zilu.cpp
@@ -16,15 +16,6 @@
 
 #define PRECISION_z
 
-/* For hipSPARSE, they use a separate complex type than for hipBLAS */
-#if defined(MAGMA_HAVE_HIP)
-  #ifdef PRECISION_z
-    #define hipblasDoubleComplex hipDoubleComplex
-  #elif defined(PRECISION_c)
-    #define hipblasComplex hipComplex
-  #endif
-#endif
-
 #if CUDA_VERSION >= 12000
    #define cusparseCreateCsrsm2Info(info)
    #define cusparseDestroyCsrsm2Info(info)
diff --git a/sparse/blas/zmergecg.cu b/sparse/blas/zmergecg.cu
index ff20497..f40154d 100644
--- a/sparse/blas/zmergecg.cu
+++ b/sparse/blas/zmergecg.cu
@@ -13,16 +13,6 @@
 
 #define PRECISION_z
 
-/* For hipSPARSE, they use a separate complex type than for hipBLAS */
-#if defined(MAGMA_HAVE_HIP)
-  #ifdef PRECISION_z
-    #define hipblasDoubleComplex hipDoubleComplex
-  #elif defined(PRECISION_c)
-    #define hipblasComplex hipComplex
-  #endif
-#endif
-
-
 #define BLOCK_SIZE 512
 
 #if CUDA_VERSION >= 12000
diff --git a/sparse/control/magma_zmconvert.cpp b/sparse/control/magma_zmconvert.cpp
index 97acef0..679b193 100644
--- a/sparse/control/magma_zmconvert.cpp
+++ b/sparse/control/magma_zmconvert.cpp
@@ -13,11 +13,6 @@
 #include <cuda.h>  // for CUDA_VERSION
 
 
-/* For hipSPARSE, they use a separate complex type than for hipBLAS */
-#ifdef MAGMA_HAVE_HIP
-  #define hipblasDoubleComplex hipDoubleComplex
-#endif
-
 
 // todo: check if we need buf later
 #if CUDA_VERSION >= 11000
diff --git a/sparse/control/magma_zmtranspose.cpp b/sparse/control/magma_zmtranspose.cpp
index 6556a9f..9366dc3 100644
--- a/sparse/control/magma_zmtranspose.cpp
+++ b/sparse/control/magma_zmtranspose.cpp
@@ -14,11 +14,6 @@
 
 #include <cuda.h>  // for CUDA_VERSION
 
-/* For hipSPARSE, they use a separate complex type than for hipBLAS */
-#ifdef MAGMA_HAVE_HIP
-  #define hipblasDoubleComplex hipDoubleComplex
-#endif
-
 // todo: check if we need buf later
 #if CUDA_VERSION >= 11000
 #define cusparseZcsr2csc(handle, m, n, nnz, valA, rowA, colA, valB, colB, rowB,                \
diff --git a/sparse/src/zcustomic.cpp b/sparse/src/zcustomic.cpp
index fa817d4..ed48f49 100644
--- a/sparse/src/zcustomic.cpp
+++ b/sparse/src/zcustomic.cpp
@@ -16,11 +16,6 @@
 
 #define COMPLEX
 
-/* For hipSPARSE, they use a separate complex type than for hipBLAS */
-#ifdef MAGMA_HAVE_HIP
-  #define hipblasDoubleComplex hipDoubleComplex
-#endif
-
 /**
     Purpose
     -------
diff --git a/sparse/src/zcustomilu.cpp b/sparse/src/zcustomilu.cpp
index d5c5613..ee990c7 100644
--- a/sparse/src/zcustomilu.cpp
+++ b/sparse/src/zcustomilu.cpp
@@ -16,11 +16,6 @@
 
 #define COMPLEX
 
-/* For hipSPARSE, they use a separate complex type than for hipBLAS */
-#ifdef MAGMA_HAVE_HIP
-  #define hipblasDoubleComplex hipDoubleComplex
-#endif
-
 /**
     Purpose
     -------
diff --git a/sparse/src/zparict.cpp b/sparse/src/zparict.cpp
index 1b40d1c..f40e7c0 100644
--- a/sparse/src/zparict.cpp
+++ b/sparse/src/zparict.cpp
@@ -21,12 +21,6 @@
 #define PRECISION_z
 
 
-/* For hipSPARSE, they use a separate complex type than for hipBLAS */
-#ifdef MAGMA_HAVE_HIP
-  #define hipblasDoubleComplex hipDoubleComplex
-#endif
-
-
 /***************************************************************************//**
     Purpose
     -------
diff --git a/sparse/src/zparilut.cpp b/sparse/src/zparilut.cpp
index 71c4025..6744f97 100644
--- a/sparse/src/zparilut.cpp
+++ b/sparse/src/zparilut.cpp
@@ -20,11 +20,6 @@
 
 #define PRECISION_z
 
-/* For hipSPARSE, they use a separate complex type than for hipBLAS */
-#ifdef MAGMA_HAVE_HIP
-  #define hipblasDoubleComplex hipDoubleComplex
-#endif
-
 /***************************************************************************//**
     Purpose
     -------
diff --git a/sparse/testing/testing_zspmm.cpp b/sparse/testing/testing_zspmm.cpp
index 3d816e9..60848c1 100644
--- a/sparse/testing/testing_zspmm.cpp
+++ b/sparse/testing/testing_zspmm.cpp
@@ -70,15 +70,6 @@
 
 #define PRECISION_z
 
-/* For hipSPARSE, they use a separate complex type than for hipBLAS */
-#if defined(MAGMA_HAVE_HIP)
-  #ifdef PRECISION_z
-    #define hipblasDoubleComplex hipDoubleComplex
-#elif defined(PRECISION_c)
-    #define hipblasComplex hipComplex
-  #endif
-#endif
-
 /* ////////////////////////////////////////////////////////////////////////////
    -- testing sparse matrix vector product
 */
diff --git a/sparse/testing/testing_zspmv.cpp b/sparse/testing/testing_zspmv.cpp
index d9e7beb..c768a32 100644
--- a/sparse/testing/testing_zspmv.cpp
+++ b/sparse/testing/testing_zspmv.cpp
@@ -72,15 +72,6 @@
 
 #define PRECISION_z
 
-/* For hipSPARSE, they use a separate complex type than for hipBLAS */
-#if defined(MAGMA_HAVE_HIP)
-  #ifdef PRECISION_z
-    #define hipblasDoubleComplex hipDoubleComplex
-#elif defined(PRECISION_c)
-    #define hipblasComplex hipComplex
-  #endif
-#endif
-
 /* ////////////////////////////////////////////////////////////////////////////
    -- testing sparse matrix vector product
 */
diff --git a/src/shpotrf_gpu.cpp b/src/shpotrf_gpu.cpp
index 00e4ccf..d3b0014 100644
--- a/src/shpotrf_gpu.cpp
+++ b/src/shpotrf_gpu.cpp
@@ -55,10 +55,10 @@ magma_sgemm_fp16(
     hipblasGemmEx( queue->hipblas_handle(),
 		           hipblas_trans_const( transA ), hipblas_trans_const( transB ),
 		           int(m), int(n), int(k),
-		           (void*)&alpha, (void*)dhA, HIPBLAS_R_16F, (int)lddha,
-                                  (void*)dhB, HIPBLAS_R_16F, (int)lddhb,
-		           (void*)&beta,  (void*)dC,  HIPBLAS_R_32F, (int)lddc,
-		           HIPBLAS_R_32F, HIPBLAS_GEMM_DEFAULT);
+		           (void*)&alpha, (void*)dhA, HIP_R_16F, (int)lddha,
+                                  (void*)dhB, HIP_R_16F, (int)lddhb,
+		           (void*)&beta,  (void*)dC,  HIP_R_32F, (int)lddc,
+		           HIPBLAS_COMPUTE_32F, HIPBLAS_GEMM_DEFAULT);
     return 0;
     #else
     return MAGMA_ERR_NOT_SUPPORTED;
diff --git a/src/xhsgetrf_gpu.cpp b/src/xhsgetrf_gpu.cpp
index b5dd4ec..d85beaa 100644
--- a/src/xhsgetrf_gpu.cpp
+++ b/src/xhsgetrf_gpu.cpp
@@ -403,10 +403,10 @@ magma_xhsgetrf_gpu(
                     hipblasGemmEx( queues[1]->hipblas_handle(),
                             hipblas_trans_const( MagmaNoTrans ), hipblas_trans_const( MagmaNoTrans ),
                             int(nextjb), int(m-nextj), int(jb),
-                            &c_neg_one, dAT_hp(j,     nextj), HIPBLAS_R_16F, int(lddat),
-                                        dAT_hp(nextj, j    ), HIPBLAS_R_16F, int(lddat),
-                            &c_one,     dAT_hp(nextj, nextj), HIPBLAS_R_16F, int(lddat),
-                            HIPBLAS_R_32F, ALGO);
+                            &c_neg_one, dAT_hp(j,     nextj), HIP_R_16F, int(lddat),
+                                        dAT_hp(nextj, j    ), HIP_R_16F, int(lddat),
+                            &c_one,     dAT_hp(nextj, nextj), HIP_R_16F, int(lddat),
+                            HIPBLAS_COMPUTE_32F, ALGO);
                     #endif
                 }
                 else if( mp_algo_type == Magma_MP_GEMEX_I16_O16_C16 ) {
@@ -422,10 +422,10 @@ magma_xhsgetrf_gpu(
                     hipblasGemmEx( queues[1]->hipblas_handle(),
                             hipblas_trans_const( MagmaNoTrans ), hipblas_trans_const( MagmaNoTrans ),
                             int(nextjb), int(m-nextj), int(jb),
-                            &h_neg_one, dAT_hp(j,     nextj), HIPBLAS_R_16F, int(lddat),
-                                        dAT_hp(nextj, j    ), HIPBLAS_R_16F, int(lddat),
-                            &h_one,     dAT_hp(nextj, nextj), HIPBLAS_R_16F, int(lddat),
-                            HIPBLAS_R_16F, ALGO);
+                            &h_neg_one, dAT_hp(j,     nextj), HIP_R_16F, int(lddat),
+                                        dAT_hp(nextj, j    ), HIP_R_16F, int(lddat),
+                            &h_one,     dAT_hp(nextj, nextj), HIP_R_16F, int(lddat),
+                            HIPBLAS_COMPUTE_16F, ALGO);
                     #endif
                 }
                 else if( mp_algo_type == Magma_MP_HGEMM ) {
@@ -480,10 +480,10 @@ magma_xhsgetrf_gpu(
                     hipblasGemmEx( queues[1]->hipblas_handle(),
                             hipblas_trans_const( MagmaNoTrans ), hipblas_trans_const( MagmaNoTrans ),
                             int(maxn-(nextj+nextjb)), int(m-nextj), int(jb),
-                            &c_neg_one, dAT_hp(j,     nextj+nextjb), HIPBLAS_R_16F, int(lddat),
-                                        dAT_hp(nextj, j           ), HIPBLAS_R_16F, int(lddat),
-                            &c_one,     dAT_hp(nextj, nextj+nextjb), HIPBLAS_R_16F, int(lddat),
-                            HIPBLAS_R_32F, ALGO);
+                            &c_neg_one, dAT_hp(j,     nextj+nextjb), HIP_R_16F, int(lddat),
+                                        dAT_hp(nextj, j           ), HIP_R_16F, int(lddat),
+                            &c_one,     dAT_hp(nextj, nextj+nextjb), HIP_R_16F, int(lddat),
+                            HIPBLAS_COMPUTE_32F, ALGO);
                     #endif
                 }
                 else if( mp_algo_type == Magma_MP_GEMEX_I16_O16_C16 ) {
@@ -499,10 +499,10 @@ magma_xhsgetrf_gpu(
                     hipblasGemmEx( queues[1]->hipblas_handle(),
                             hipblas_trans_const( MagmaNoTrans ), hipblas_trans_const( MagmaNoTrans ),
                             int(maxn-(nextj+nextjb)), int(m-nextj), int(jb),
-                            &h_neg_one, dAT_hp(j,     nextj+nextjb), HIPBLAS_R_16F, int(lddat),
-                                        dAT_hp(nextj, j           ), HIPBLAS_R_16F, int(lddat),
-                            &h_one,     dAT_hp(nextj, nextj+nextjb), HIPBLAS_R_16F, int(lddat),
-                            HIPBLAS_R_16F, ALGO);
+                            &h_neg_one, dAT_hp(j,     nextj+nextjb), HIP_R_16F, int(lddat),
+                                        dAT_hp(nextj, j           ), HIP_R_16F, int(lddat),
+                            &h_one,     dAT_hp(nextj, nextj+nextjb), HIP_R_16F, int(lddat),
+                            HIPBLAS_COMPUTE_16F, ALGO);
                     #endif
                 }
                 else if( mp_algo_type == Magma_MP_HGEMM ) {
diff --git a/src/xshgetrf_gpu.cpp b/src/xshgetrf_gpu.cpp
index 63d0da6..a3a6343 100644
--- a/src/xshgetrf_gpu.cpp
+++ b/src/xshgetrf_gpu.cpp
@@ -270,10 +270,10 @@ magma_xshgetrf_gpu(
                     hipblasGemmEx( queues[1]->hipblas_handle(),
 		                    hipblas_trans_const( MagmaNoTrans ), hipblas_trans_const( MagmaNoTrans ),
 		                    int(nextjb), int(m-nextj), int(jb),
-		                    &c_neg_one, dAtrsm1_hp,        HIPBLAS_R_16F, int(maxnb),
-		                                dApanel_hp,        HIPBLAS_R_16F, int(jb),
-		                    &c_one,     dAT(nextj, nextj), HIPBLAS_R_32F, int(lddat),
-		                    HIPBLAS_R_32F, ALGO);
+		                    &c_neg_one, dAtrsm1_hp,        HIP_R_16F, int(maxnb),
+		                                dApanel_hp,        HIP_R_16F, int(jb),
+		                    &c_one,     dAT(nextj, nextj), HIP_R_32F, int(lddat),
+		                    HIPBLAS_COMPUTE_32F, ALGO);
                     #endif
                 }
                 else if( mp_algo_type == Magma_MP_GEMEX_I32_O32_C32 ) {
@@ -289,10 +289,10 @@ magma_xshgetrf_gpu(
                     hipblasGemmEx( queues[1]->hipblas_handle(),
                             hipblas_trans_const( MagmaNoTrans ), hipblas_trans_const( MagmaNoTrans ),
                             int(nextjb), int(m-nextj), int(jb),
-                            &c_neg_one, dAT(j,     nextj), HIPBLAS_R_32F, int(lddat),
-                                        dAT(nextj,     j), HIPBLAS_R_32F, int(lddat),
-                            &c_one,     dAT(nextj, nextj), HIPBLAS_R_32F, int(lddat),
-                            HIPBLAS_R_32F, ALGO);
+                            &c_neg_one, dAT(j,     nextj), HIP_R_32F, int(lddat),
+                                        dAT(nextj,     j), HIP_R_32F, int(lddat),
+                            &c_one,     dAT(nextj, nextj), HIP_R_32F, int(lddat),
+                            HIPBLAS_COMPUTE_32F, ALGO);
                     #endif
                 }
                 else if( mp_algo_type == Magma_MP_SGEMM ) {
@@ -343,10 +343,10 @@ magma_xshgetrf_gpu(
                     hipblasGemmEx( queues[1]->hipblas_handle(),
                             hipblas_trans_const( MagmaNoTrans ), hipblas_trans_const( MagmaNoTrans ),
                             int(maxn-(nextj+nextjb)), int(m-nextj), int(jb),
-                            &c_neg_one, dAtrsm2_hp              , HIPBLAS_R_16F, int(maxm),
-                                        dApanel_hp              , HIPBLAS_R_16F, int(jb),
-                            &c_one,     dAT(nextj, nextj+nextjb), HIPBLAS_R_32F, int(lddat),
-                            HIPBLAS_R_32F, ALGO);
+                            &c_neg_one, dAtrsm2_hp              , HIP_R_16F, int(maxm),
+                                        dApanel_hp              , HIP_R_16F, int(jb),
+                            &c_one,     dAT(nextj, nextj+nextjb), HIP_R_32F, int(lddat),
+                            HIPBLAS_COMPUTE_32F, ALGO);
                     #endif
                 }
                 else if( mp_algo_type == Magma_MP_GEMEX_I32_O32_C32 ) {
@@ -362,10 +362,10 @@ magma_xshgetrf_gpu(
                     hipblasGemmEx( queues[1]->hipblas_handle(),
                             hipblas_trans_const( MagmaNoTrans ), hipblas_trans_const( MagmaNoTrans ),
                             int(maxn-(nextj+nextjb)), int(m-nextj), int(jb),
-                            &c_neg_one, dAT(j    , nextj+nextjb), HIPBLAS_R_32F, int(lddat),
-                                        dAT(nextj, j           ), HIPBLAS_R_32F, int(lddat),
-                            &c_one,     dAT(nextj, nextj+nextjb), HIPBLAS_R_32F, int(lddat),
-                            HIPBLAS_R_32F, ALGO);
+                            &c_neg_one, dAT(j    , nextj+nextjb), HIP_R_32F, int(lddat),
+                                        dAT(nextj, j           ), HIP_R_32F, int(lddat),
+                            &c_one,     dAT(nextj, nextj+nextjb), HIP_R_32F, int(lddat),
+                            HIPBLAS_COMPUTE_32F, ALGO);
                     #endif
                 }
                 else if( mp_algo_type == Magma_MP_SGEMM ) {
diff --git a/testing/testing_sgemm_fp16.cpp b/testing/testing_sgemm_fp16.cpp
index b9fdabd..8c1e870 100644
--- a/testing/testing_sgemm_fp16.cpp
+++ b/testing/testing_sgemm_fp16.cpp
@@ -59,10 +59,10 @@ magma_sgemm_fp16_v1(
     hipblasGemmEx( magma_queue_get_hipblas_handle( queue ),
 		           hipblas_trans_const( transA ), hipblas_trans_const( transB ),
 		           int(m), int(n), int(k),
-		           (void*)&alpha, (void*)dhA, HIPBLAS_R_16F, (int)ldda,
-                                  (void*)dhB, HIPBLAS_R_16F, (int)lddb,
-		           (void*)&beta,  (void*)dC,  HIPBLAS_R_32F, (int)lddc,
-		           HIPBLAS_R_32F, HIPBLAS_GEMM_DEFAULT);
+		           (void*)&alpha, (void*)dhA, HIP_R_16F, (int)ldda,
+                                  (void*)dhB, HIP_R_16F, (int)lddb,
+		           (void*)&beta,  (void*)dC,  HIP_R_32F, (int)lddc,
+		           HIPBLAS_COMPUTE_32F, HIPBLAS_GEMM_DEFAULT);
     #endif
     return 0;
 }
@@ -97,10 +97,10 @@ magma_sgemm_fp16_v2(
     hipblasGemmEx( magma_queue_get_hipblas_handle( queue ),
 		           hipblas_trans_const( transA ), hipblas_trans_const( transB ),
 		           int(m), int(n), int(k),
-		           (void*)&alpha, (void*)dhA, HIPBLAS_R_16F, (int)ldda,
-                                  (void*)dhB, HIPBLAS_R_16F, (int)lddb,
-		           (void*)&beta,  (void*)dC,  HIPBLAS_R_32F, (int)lddc,
-		           HIPBLAS_R_32F, HIPBLAS_GEMM_DEFAULT);
+		           (void*)&alpha, (void*)dhA, HIP_R_16F, (int)ldda,
+                                  (void*)dhB, HIP_R_16F, (int)lddb,
+		           (void*)&beta,  (void*)dC,  HIP_R_32F, (int)lddc,
+		           HIPBLAS_COMPUTE_32F, HIPBLAS_GEMM_DEFAULT);
     #endif
     return 0;
 }
diff --git a/testing/testing_zgemm_batched.cpp b/testing/testing_zgemm_batched.cpp
index 8b6ddbd..dbddee9 100644
--- a/testing/testing_zgemm_batched.cpp
+++ b/testing/testing_zgemm_batched.cpp
@@ -190,11 +190,11 @@ int main( int argc, char** argv)
                   hipblasZgemmBatched(
                                    opts.handle, cublas_trans_const(opts.transA), cublas_trans_const(opts.transB),
                                    int(M), int(N), int(K),
-                                   (const hipblasDoubleComplex*)&alpha,
-                                   (const hipblasDoubleComplex**) d_A_array, int(ldda),
-                                   (const hipblasDoubleComplex**) d_B_array, int(lddb),
-                                   (const hipblasDoubleComplex*)&beta,
-                                   (hipblasDoubleComplex**)d_C_array, int(lddc), int(batchCount) );
+                                   (const hipDoubleComplex*)&alpha,
+                                   (const hipDoubleComplex**) d_A_array, int(ldda),
+                                   (const hipDoubleComplex**) d_B_array, int(lddb),
+                                   (const hipDoubleComplex*)&beta,
+                                   (hipDoubleComplex**)d_C_array, int(lddc), int(batchCount) );
                 #endif
             }
             else{
@@ -211,11 +211,11 @@ int main( int argc, char** argv)
                 hipblasZgemmStridedBatched(
                                    opts.handle, cublas_trans_const(opts.transA), cublas_trans_const(opts.transB),
                                    int(M), int(N), int(K),
-                                   (const hipblasDoubleComplex*)&alpha,
-                                   (const hipblasDoubleComplex*) d_A, int(ldda), ldda * An,
-                                   (const hipblasDoubleComplex*) d_B, int(lddb), lddb * Bn,
-                                   (const hipblasDoubleComplex*)&beta,
-                                   (hipblasDoubleComplex*)d_C, int(lddc), lddc*N, int(batchCount) );
+                                   (const hipDoubleComplex*)&alpha,
+                                   (const hipDoubleComplex*) d_A, int(ldda), ldda * An,
+                                   (const hipDoubleComplex*) d_B, int(lddb), lddb * Bn,
+                                   (const hipDoubleComplex*)&beta,
+                                   (hipDoubleComplex*)d_C, int(lddc), lddc*N, int(batchCount) );
                 #endif
             }
 
diff --git a/testing/testing_zgemv_batched.cpp b/testing/testing_zgemv_batched.cpp
index f7fabc8..fc7422e 100644
--- a/testing/testing_zgemv_batched.cpp
+++ b/testing/testing_zgemv_batched.cpp
@@ -182,11 +182,11 @@ int main( int argc, char** argv)
                 #else
                 hipblasZgemvBatched(opts.handle, hipblas_trans_const(opts.transA),
                                       M, N,
-                                      (const hipblasDoubleComplex *)&alpha,
-                                      (const hipblasDoubleComplex **)d_A_array, ldda,
-                                      (const hipblasDoubleComplex **)d_X_array, incx,
-                                      (const hipblasDoubleComplex *)&beta,
-                                      (hipblasDoubleComplex **)d_Y_array, incy, batchCount);
+                                      (const hipDoubleComplex *)&alpha,
+                                      (const hipDoubleComplex **)d_A_array, ldda,
+                                      (const hipDoubleComplex **)d_X_array, incx,
+                                      (const hipDoubleComplex *)&beta,
+                                      (hipDoubleComplex **)d_Y_array, incy, batchCount);
                 #endif
             }
             else{
@@ -210,11 +210,11 @@ int main( int argc, char** argv)
                 #else
                 hipblasZgemvStridedBatched(opts.handle, hipblas_trans_const(opts.transA),
                                       M, N,
-                                      (const hipblasDoubleComplex *)&alpha,
-                                      (const hipblasDoubleComplex *)d_A, ldda, ldda*N,
-                                      (const hipblasDoubleComplex *)d_X, incx, incx*Xm,
-                                      (const hipblasDoubleComplex *)&beta,
-                                      (hipblasDoubleComplex *)d_Y, incy, incy*Ym, batchCount);
+                                      (const hipDoubleComplex *)&alpha,
+                                      (const hipDoubleComplex *)d_A, ldda, ldda*N,
+                                      (const hipDoubleComplex *)d_X, incx, incx*Xm,
+                                      (const hipDoubleComplex *)&beta,
+                                      (hipDoubleComplex *)d_Y, incy, incy*Ym, batchCount);
                 #endif
             }
             device_time = magma_sync_wtime( opts.queue ) - device_time;
diff --git a/testing/testing_zgeqrf_batched.cpp b/testing/testing_zgeqrf_batched.cpp
index 0364401..eea910b 100644
--- a/testing/testing_zgeqrf_batched.cpp
+++ b/testing/testing_zgeqrf_batched.cpp
@@ -195,8 +195,8 @@ int main( int argc, char** argv)
             #endif
             #else
             hipblasZgeqrfBatched( opts.handle, int(M), int(N),
-                                 (hipblasDoubleComplex**)dA_array, int(ldda),
-                                 (hipblasDoubleComplex**)dtau_array,
+                                 (hipDoubleComplex**)dA_array, int(ldda),
+                                 (hipDoubleComplex**)dtau_array,
                                  &device_info, int(batchCount) );
             #endif
 
diff --git a/testing/testing_zgetrf_batched.cpp b/testing/testing_zgetrf_batched.cpp
index 706de5b..d3a7ada 100644
--- a/testing/testing_zgetrf_batched.cpp
+++ b/testing/testing_zgetrf_batched.cpp
@@ -238,7 +238,7 @@ int main( int argc, char** argv)
                                      dinfo_device, int(batchCount) );
                 #else
                 hipblasZgetrfBatched( opts.handle, int(N),
-                                     (hipblasDoubleComplex**)dA_array, int(ldda), dipiv_device,
+                                     (hipDoubleComplex**)dA_array, int(ldda), dipiv_device,
                                      dinfo_device, int(batchCount) );
                 #endif
             }
diff --git a/testing/testing_ztrsm_batched.cpp b/testing/testing_ztrsm_batched.cpp
index 6c092fd..3c850d5 100644
--- a/testing/testing_ztrsm_batched.cpp
+++ b/testing/testing_ztrsm_batched.cpp
@@ -219,9 +219,9 @@ int main( int argc, char** argv)
             hipblasZtrsmBatched(
                 opts.handle, cublas_side_const(opts.side), cublas_uplo_const(opts.uplo),
                 cublas_trans_const(opts.transA), cublas_diag_const(opts.diag),
-                int(M), int(N), (const hipblasDoubleComplex*)&alpha,
-                (hipblasDoubleComplex* const*) d_A_array, int(ldda),
-                (      hipblasDoubleComplex**) d_B_array, int(lddb), int(batchCount) );
+                int(M), int(N), (const hipDoubleComplex*)&alpha,
+                (hipDoubleComplex* const*) d_A_array, int(ldda),
+                (      hipDoubleComplex**) d_B_array, int(lddb), int(batchCount) );
             #endif
 
             cublas_time = magma_sync_wtime( opts.queue ) - cublas_time;
