diff --git a/CMakeLists.txt b/CMakeLists.txt
index c50bcb63e..c4684ca72 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -289,7 +289,7 @@ if (MAGMA_ENABLE_HIP)
          set( DEVCCFLAGS  "" )
-    set(VALID_GFXS "700;701;702;703;704;705;801;802;803;805;810;900;902;904;906;908;909;90c;1010;1011;1012;1030;1031;1032;1033")
+    set(VALID_GFXS "700;701;702;703;704;705;801;802;803;805;810;900;902;904;906;908;909;90c;1010;1011;1012;1030;1031;1032;1033;1150;1151")
     foreach( GFX ${VALID_GFXS} )
       if ( GPU_TARGET MATCHES gfx${GFX} )
         set( DEVCCFLAGS ${DEVCCFLAGS} --offload-arch=gfx${GFX} )
